crates/sncast/src/starknet_commands/script.rs

```diff
@@ -37,14 +39,182 @@ use sncast::helpers::configuration::CastConfig;
 use sncast::helpers::constants::SCRIPT_LIB_ARTIFACT_NAME;
 use sncast::response::print::print_as_warning;
 use sncast::response::structs::ScriptResponse;
+use sncast::{ErrorData, TransactionError};
 use starknet::accounts::Account;
+use starknet::core::types::StarknetError;
 use starknet::core::types::{BlockId, BlockTag::Pending, FieldElement};
 use starknet::providers::jsonrpc::HttpTransport;
-use starknet::providers::JsonRpcClient;
+use starknet::providers::{JsonRpcClient, ProviderError};
 use tokio::runtime::Runtime;
 
 type ScriptStarknetContractArtifacts = StarknetContractArtifacts;
 
+pub trait SerializeAsFelt252 {
+    fn serialize_as_felt252(&self) -> Vec<Felt252>;
+}
```

Comment:
We can also implement this for `crates/sncast/src/response/structs.rs`, simplifying script subcommand Result serialization https://github.com/foundry-rs/starknet-foundry/pull/1705/files#diff-3623497f34cf1bd0ededf31f95e511abb9ae5b8791f80cce7dd7e8e066031ab6R298-R311

To limit the scope of this PR I suggest doing it in a separate PR

Reply:
We can put introduced errors and `SerializeAsFelt252` trait implementations into a separate file, like `script/response.rs`

Reply:
> We can also implement this for crates/sncast/src/response/structs.rs and Result<T, E>, simplifying script subcommand Result serialization
To limit the scope of this PR I suggest doing it in a separate PR

Sounds like a good idea

> We can put introduced errors and SerializeAsFelt252 trait implementations into a separate file, like script/response.rs

I'd suggest so, yeah. We could even probably put this to src/response/script.rs, to keep all response-related things in one place

Reply:
https://github.com/foundry-rs/starknet-foundry/issues/1721